### PR TITLE
feat(rest): Added endpoint to get changelog by document id

### PR DESCRIPTION
**Pull request by JaideepPalit**
>  Added endpoint to get changelog by document id
> * Which issue is this pull request belonging to and how is it solving it? (*#1090*)
> * Did you add or update any new dependencies that are required for your change? - No

   - Execute Endpoint `/api/changelog/document/{docId}?page=0&page_entries=5&sort=changeTimestamp%2Casc` with pagination parameter
> Have you implemented any additional tests?

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>

### CODE REVIEW

**Review by smrutis1 — CHANGES_REQUESTED** [review]
Few Observations:

* Please add about change log in section `3. Resources` of documentation.
* In pagination field description please add appropriate field description instead of search results. ref: 
image


This endpoint provides list of changelogs of a document. Similarly I think another endpoint would be good to fetch individual changelog by using  changelog id.

**Comment by JaideepPalit** [comment]
Thanks for review and test.
Made changes as suggested.

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ChangeLogSpecTest.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                                fieldWithPath("_embedded.sw360:changeLogs[]referenceDoc[]refDocType").description("The type of the reference document. Possible values are " + docType),
+                                fieldWithPath("_embedded.sw360:changeLogs[]referenceDoc[]refDocOperation").description("The type of the operation. Possible values are " + Arrays.asList(Operation.values())),
+                                fieldWithPath("_embedded.sw360:changeLogs[]referenceDoc[]").description("An array of refernce doc"),
+                                fieldWithPath("_embedded.sw360:changeLogs").description("An array of <<resources-changelog, ChangeLog resources>>"),
+                                fieldWithPath("_links").description("<<resources-index-links,Links>> to other resources"),
+                                fieldWithPath("page").description("Additional paging information"),
+                                fieldWithPath("page.size").description("Number of search results per page"),
```
Comment (smrutis1) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ChangeLogSpecTest.java:164]: Please update search results to changelogs here and in below line too

**Review by smrutis1 — CHANGES_REQUESTED** [review]
Seems like in HAL browser the highlighted link in red gives an `404` response. and doc link here is also seems to be not working.

image

image

**Comment by JaideepPalit** [comment]
Added fix for above comment

**Review by smrutis1 — APPROVED** [review]
Code looks good to me. Tested successfully.

**Pull request merged**